WordPress/Classes/ViewRelated/Reader/Filter/FilterProvider.swift

```diff
@@ -49,12 +51,18 @@ class FilterProvider: Observable, FilterTabBarItem {
          accessibilityIdentifier: String,
          cellClass: UITableViewCell.Type,
          reuseIdentifier: String,
+         emptyTitle: String,
+         emptyActionTitle: String,
+         emptyAction: (@escaping (UIViewController) -> Void),
          provider: (@escaping (Result<[TableDataItem], Error>) -> Void) -> Void) {
 
         titleFunc = title
         self.accessibilityIdentifier = accessibilityIdentifier
         self.cellClass = cellClass
         self.reuseIdentifier = reuseIdentifier
+        self.emptyTitle = emptyTitle
+        self.emptyActionTitle = emptyActionTitle
+        self.emptyAction = emptyAction
```

Comment:
I wonder if we could just pass a `TabbedSection` value here and from that, the view could handle the titles, and maybe we could define define `showAdd(_:)` directly in `FilterProvider` instead of two separate extensions?

Reply:
This was a great suggestion! Much cleaner now 👍